### ISSUE TITLE
Move GPU Enabling to a subscript

### ISSUE THREAD

**Issue by wildintellect** [open-forest-observatory/automate-metashape#7]
The code to check for GPUs, report their names and enable if available or configured can be it's own script.

Configuration should include
GPU_enable = True/False
GPU_num = numeric (number of GPUs to use if available)

**Comment by youngdjn** [comment]
This is now a function defined in metashape_pipeline_functions.py. It can be included/excluded as a call from a control script as a means of enabling/not enabling. I have not yet implemented GPU_num because I could not immediately see how to specify the number. Do you feel the "GPU subscript" component of this issue is addressed? If so, we can close this and create a separate ticket for the GPU_num issue.

**Comment by wildintellect** [comment]
We do need to modify this if we plan to ever use multiple GPUs.

> Application.gpu_maskGPU device bit mask: 1 - use device, 0 - do not use (i.e. value 5 enables device number 0 and 2).

A small function to translate from an optional number, default=1 to the appropriate bitmask will do.

**Comment by youngdjn** [comment]
Will this require each parallel instance of the script to know which specific GPUs the other instances are using?

**Comment by wildintellect** [comment]
No, since the resource sets the number of GPUs available they will always be labeled 1 to n. So typically we would use all available, so the user can just provide n (or slurm as an env variable might) and we translate to enable all for however many n that is.
